src/modules/item/sagas.ts

```diff
@@ -87,6 +87,7 @@ export function* itemSaga() {
   yield takeEvery(FETCH_COLLECTION_REQUEST, handleFetchCollectionRequest)
   yield takeLatest(FETCH_TRANSACTION_SUCCESS, handleTransactionSuccess)
 
+  // TODO: Why do we have take latest here? What if there are multiple failures waiting to be handled?
```

Comment:
Why do we take latest here?

Reply:
shouldn't it be the same? I mean, we are trying to fix it just once so handling multiple failure states should be the same as just handling the last one

Reply:
If we use latest, how do you handle two deploy or set items that need to be retried?
What if you load two collections (by doing a refresh of the site) that have been created in the blockchain but their items don't have their ids or their content wasn't deployed to the Catalyst. As far as I know, the post-processing of those elements will be done at the same time. If for example, the set items id request fails for both of them it could happen that only one of them is retried.
Also, there's no case were a failure action would appear more than once because the retrying mechanism happens only when one of those actions is handled.

Reply:
You're correct! In my mind this FAILURE case would retry _all collections/items_ and not the one that errored out. I think this should be a `takeEvery`